src/verify.c

```diff
@@ -993,20 +1089,42 @@ enum swupd_code verify_main(void)
 		}
 	}
 
-	set_subscription_versions(official_manifest, NULL, &subs);
-	official_manifest->submanifests = recurse_manifest(official_manifest, subs, NULL, false, NULL);
-	if (!official_manifest->submanifests) {
+	set_subscription_versions(official_manifest, NULL, &all_subs);
+	set_subscription_versions(official_manifest, NULL, &bundles_subs);
+	all_submanifests = recurse_manifest(official_manifest, all_subs, NULL, false, NULL);
+	if (!all_submanifests) {
 		error("Cannot load MoM sub-manifests\n");
 		ret = SWUPD_RECURSE_MANIFEST;
 		goto clean_and_exit;
 	}
+	official_manifest->submanifests = all_submanifests;
+	if (cmdline_option_bundles) {
+		bundles_submanifests = recurse_manifest(official_manifest, bundles_subs, NULL, false, NULL);
+		if (!bundles_submanifests) {
+			error("Cannot load MoM sub-manifests\n");
+			ret = SWUPD_RECURSE_MANIFEST;
+			goto clean_and_exit;
+		}
+		official_manifest->submanifests = bundles_submanifests;
+	}
 	progress_complete_step();
 	timelist_timer_stop(globals.global_times); // closing: Load manifests
 
 	timelist_timer_start(globals.global_times, "Consolidate files from bundles");
 	progress_set_step(4, "consolidate_files");
-	official_manifest->files = files_from_bundles(official_manifest->submanifests);
-	official_manifest->files = consolidate_files(official_manifest->files);
+	all_files = files_from_bundles(all_submanifests);
```

Comment:
this block of calls files_from_bundles() + consolidate_files() can be consolidated in a function like consolidated_files_from_bundles().

Reply:
Nice, I'll do that